File: web/src/types/analytics.ts
export type ReportType = 'daily_summary' | 'device_efficiency' | 'item_distribution' | 'wait_trend' | 'no_show'
export type ReportStatus = 'generating' | 'ready' | 'failed'
export type ReportFormat = 'xlsx' | 'pdf' | 'csv'
export type AlertLevel = 'info' | 'warning' | 'critical'

export interface SlotUsage {
  total_slots: number
  used_slots: number
  available_slots: number
  usage_rate: number
}

export interface DeviceStatusItem {
  device_id: string
  device_name: string
  status: 'idle' | 'busy' | 'offline'
  queue_count: number
}

export interface WaitTrendPoint {
  time: string
  avg_wait_min: number
}

export interface DashboardAlert {
  type: AlertLevel
  message: string
  device_id?: string
  value?: number
}

export interface DashboardSnapshot {
  id: string
  timestamp: string
  slot_usage: SlotUsage
  device_status: DeviceStatusItem[]
  wait_trend: WaitTrendPoint[]
  alerts: DashboardAlert[]
}

export interface DeviceDetail {
  device_id: string
  device_name: string
  date: string
  total_exams: number
  avg_duration_min: number
  utilization_rate: number
  hourly_breakdown: Array<{ hour: number, count: number, avg_wait_min: number }>
}

export interface Report {
  id: string
  report_type: ReportType
  dimensions?: Record<string, unknown>
  date_start: string
  date_end: string
  status: ReportStatus
  file_path?: string
  file_size?: number
  format: ReportFormat
  generated_at?: string
  created_at: string
}

export interface GenerateReportRequest {
  report_type: ReportType
  date_start: string
  date_end: string
  dimensions?: Record<string, unknown>
  format?: ReportFormat
}

File: src/internal/infrastructure/persistence/po/analytics.go
// Package po 基础设施层 - analytics 持久化对象
package po

import "time"

// DashboardSnapshotPO 大屏快照持久化对象
type DashboardSnapshotPO struct {
	ID        string    `gorm:"primaryKey;column:id;size:36"`
	CampusID  string    `gorm:"column:campus_id;size:36;index:idx_dashboard_campus_time"`
	Snapshot  string    `gorm:"column:snapshot;type:text"` // JSON
	CreatedAt time.Time `gorm:"column:created_at;autoCreateTime;index:idx_dashboard_campus_time"`
}

func (DashboardSnapshotPO) TableName() string { return "dashboard_snapshots" }

// ReportPO 报表持久化对象
type ReportPO struct {
	ID          string     `gorm:"primaryKey;column:id;size:36"`
	ReportType  string     `gorm:"column:report_type;size:10;not null"`
	Dimensions  string     `gorm:"column:dimensions;type:text"` // JSON
	DateStart   time.Time  `gorm:"column:date_start;not null"`
	DateEnd     time.Time  `gorm:"column:date_end;not null"`
	Status      string     `gorm:"column:status;size:15;not null;default:generating"`
	Format      string     `gorm:"column:format;size:5;not null;default:xlsx"`
	FilePath    string     `gorm:"column:file_path;size:500"`
	FileSize    int64      `gorm:"column:file_size;default:0"`
	GeneratedAt *time.Time `gorm:"column:generated_at"`
	CreatedBy   string     `gorm:"column:created_by;size:36;not null"`
	CreatedAt   time.Time  `gorm:"column:created_at;autoCreateTime"`
}

func (ReportPO) TableName() string { return "reports" }
